httpie/models.py

```diff
@@ -45,11 +46,44 @@ def content_type(self) -> str:
 class HTTPResponse(HTTPMessage):
     """A :class:`requests.models.Response` wrapper."""
 
+    def __init__(self, orig):
+        super().__init__(orig)
+
+        # Store the first chunk of the future body to be able to get it and detect its encoding.
+        # Without such mechanism, as soon as the body will be fetched, it will
+        # be impossible to detect its encoding because the response will have
+        # consumed all the content.
+        self._first_chunk_body = BODY_NOT_SET
+
+        # Let's cache the encoding for performances concerns.
+        self._encoding = None
+
+    def _set_and_get_body_from_iterator(self, it):
+        """Set the first chunk body from the given ``it`` iterator.
+        Return the chunk for convenience.
+
+        """
+        try:
+            self._first_chunk_body = []
+            for _ in range(5):
+                self._first_chunk_body.append(next(it))
+        except StopIteration:
+            pass
+        return self._first_chunk_body
```

Comment:
this breaks streaming

Reply:
I know and I am trying different approaches, nothing really working for now.
Also I reviewed that part with d826da219acd92f0f5b18b28675f1f2ef221fbcc to not break streaming when the encoding was effectively already detected. That lowers a little bit the impact but still break streaming for other use cases.